maps/rivers_and_bridges.go

```diff
@@ -374,3 +376,147 @@ var riversAndBridgesExtraLargeHazards = []rules.Point{
+		Author:      "Battlesnake",
+		Version:     1,
+		MinPlayers:  1,
+		MaxPlayers:  4,
+		BoardSizes:  FixedSizes(Dimensions{11, 11}),
+		Tags:        []string{TAG_FOOD_PLACEMENT, TAG_HAZARD_PLACEMENT, TAG_SNAKE_PLACEMENT},
+	}
+}
+
+func (m IslandsAndBridgesMediumHazardsMap) SetupBoard(lastBoardState *rules.BoardState, settings rules.Settings, editor Editor) error {
+	if !m.Meta().BoardSizes.IsAllowable(lastBoardState.Width, lastBoardState.Height) {
+		return rules.RulesetError("This map can only be played on a 11x11 board")
+	}
+
+	err := setupRiverAndBridgesBoard(islandsAndBridgesMediumStartPositions, islandsAndBridgesMediumHazards, lastBoardState, settings, editor)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (m IslandsAndBridgesMediumHazardsMap) UpdateBoard(lastBoardState *rules.BoardState, settings rules.Settings, editor Editor) error {
+	return placeRiverAndBridgesFood(lastBoardState, settings, editor)
+}
+
+var islandsAndBridgesMediumStartPositions = [][]rules.Point{
+	{
+		{X: 3, Y: 1}, {X: 1, Y: 3},
+	},
+	{
+		{X: 9, Y: 7}, {X: 7, Y: 9},
+	},
+	{
+		{X: 3, Y: 9}, {X: 1, Y: 7},
+	},
+	{
+		{X: 7, Y: 1}, {X: 9, Y: 3},
+	},
+}
+
+var islandsAndBridgesMediumHazards = []rules.Point{
+	{X: 5, Y: 10},
+	{X: 5, Y: 9},
+	{X: 5, Y: 7},
+	{X: 5, Y: 6},
+	{X: 5, Y: 5},
+	{X: 5, Y: 4},
+	{X: 5, Y: 3},
+	{X: 5, Y: 0},
+	{X: 5, Y: 1},
+	{X: 6, Y: 5},
+	{X: 7, Y: 5},
+	{X: 9, Y: 5},
+	{X: 10, Y: 5},
+	{X: 4, Y: 5},
+	{X: 3, Y: 5},
+	{X: 1, Y: 5},
+	{X: 0, Y: 5},
+	{X: 1, Y: 10},
+	{X: 9, Y: 10},
+	{X: 1, Y: 0},
+	{X: 9, Y: 0},
+	{X: 10, Y: 1},
+	{X: 10, Y: 0},
+	{X: 10, Y: 10},
+	{X: 10, Y: 9},
+	{X: 0, Y: 10},
+	{X: 0, Y: 9},
+	{X: 0, Y: 1},
+	{X: 0, Y: 0},
+	{X: 0, Y: 6},
+	{X: 0, Y: 4},
+	{X: 10, Y: 6},
+	{X: 10, Y: 4},
+	{X: 6, Y: 10},
+	{X: 4, Y: 10},
+	{X: 6, Y: 0},
+	{X: 4, Y: 0},
+}
+
+type IslandsAndBridgesLargeHazardsMap struct{}
+
+func (m IslandsAndBridgesLargeHazardsMap) ID() string {
+	return "hz_islands_bridges_lg"
+}
+
+func (m IslandsAndBridgesLargeHazardsMap) Meta() Metadata {
+	return Metadata{
+		Name:        "hz_islands_bridges_lg",
+		Description: `Creates fixed maps that have a lake of hazard in the middle with rivers going in the cardinal directions and around the edges of the map. Bridges across the rivers are provided at key points`,
+		Author:      "Battlesnake",
+		Version:     1,
+		MinPlayers:  1,
+		MaxPlayers:  16,
+		BoardSizes:  FixedSizes(Dimensions{19, 19}),
+		Tags:        []string{TAG_FOOD_PLACEMENT, TAG_HAZARD_PLACEMENT, TAG_SNAKE_PLACEMENT},
+	}
+}
+
+func (m IslandsAndBridgesLargeHazardsMap) SetupBoard(lastBoardState *rules.BoardState, settings rules.Settings, editor Editor) error {
+	if !m.Meta().BoardSizes.IsAllowable(lastBoardState.Width, lastBoardState.Height) {
+		return rules.RulesetError("This map can only be played on a 11x11 board")
```

Comment:
Typo in the error message here: should be 19x19.

I'm starting to think this could just be a helper that returns an error since we're duplicating it and the required dimensions everywhere. Something like:
```go
if err := rules.CheckMapSize(m.Meta().BoardSizes, lastBoardState.Width, lastBoardState.Height); err != nil {
	return err
}
```

Reply:
agreed

Reply:
fixed